spec/lib/site_settings/type_supervisor_spec.rb

```diff
@@ -327,6 +341,31 @@ def self.schema
              SiteSetting.types[:string],
            ]
       end
+
+      it "raises when an object is not valid for the given schema" do
+        expect {
+          settings.type_supervisor.to_db_value(:type_objects, "not-json")
+        }.to raise_error Discourse::InvalidParameters
+      end
+
+      it "raises when an object property is not valid for the given schema" do
+        expect {
+          settings.type_supervisor.to_db_value(:type_objects, "[{\"nam\":\"Brett\"}]")
+        }.to raise_error Discourse::InvalidParameters
+      end
+
+      it "raises when an object value is not valid for the given schema" do
+        expect {
+          settings.type_supervisor.to_db_value(:type_objects, "[{\"name\":1}]")
+        }.to raise_error Discourse::InvalidParameters
+      end
+
+      it "returns value for the given objects schema string setting" do
+        expect(settings.type_supervisor.to_db_value(:type_objects, "[]")).to eq [
+             "[]",
+             SiteSetting.types[:objects],
+           ]
```

Comment:
I think we should probably test with a correct object instead of just an empty value.